crud.py

```diff
@@ -0,0 +1,53 @@
+from sqlalchemy import select
+from sqlalchemy.orm import Session
+
+import schemas
+from db import models
+
+
+def get_all_authors_paginated(db: Session, page: int = 0, size: int = 10):
+    offset = page * size
+    selected_authors = select(models.Author).offset(offset).limit(size)
+    return db.execute(selected_authors).scalars().all()
+
+
+def get_all_books_paginated(
+        db: Session, author_id: int, page: int = 0, size: int = 10
+):
+    queryset = db.query(models.Book)
+    if author_id is not None:
```

Comment:
The check `if author_id is not None:` is redundant since `author_id` is a required parameter. You can remove this check.